### PR TITLE
Bug/agent loses controls after failed transfer

### PR DESCRIPTION
**Pull request by cbbaker**

### CODE REVIEW

**Comment by jhl-zest** [comment]
lgtm.   i recall there was a potential case of pulling up old calls that were not in terminated state - did we verify that this was okay?   still a bit unsure about that case -

File: spec/fixtures/vcr_cassettes/Reloading_the_widget_during_a_call/_create/during_a_call/after_a_failed_transfer/pushes_the_event_for_the_active_call.yml [hunk 1]
```diff
@@ -0,0 +1,936 @@
+---
+http_interactions:
+- request:
+    method: post
+    uri: https://AC8ab65d320fc5ca22b757d937ef013a31:c63af64b012e06a4fa4bf6223823c834@api.twilio.com/2010-04-01/Accounts/AC8ab65d320fc5ca22b757d937ef013a31/Calls/sid-4.json
```
Comment (josephcruz) [inline spec/fixtures/vcr_cassettes/Reloading_the_widget_during_a_call/_create/during_a_call/after_a_failed_transfer/pushes_the_event_for_the_active_call.yml:5]: Does this have to be obfuscated?

**Comment by cbbaker** [comment]
BJB and I tested with an agent who had unterminated calls. The 'unless agent.on_a_call?' code deals with that case.

File: spec/fixtures/vcr_cassettes/Reloading_the_widget_during_a_call/_create/during_a_call/after_a_failed_transfer/pushes_the_event_for_the_active_call.yml (hunk 1 above)
Comment (cbbaker) [inline spec/fixtures/vcr_cassettes/Reloading_the_widget_during_a_call/_create/during_a_call/after_a_failed_transfer/pushes_the_event_for_the_active_call.yml:5]: This file is generated by VCR. I don't know if there's any way to improve it.

Comment (josephcruz) [inline spec/fixtures/vcr_cassettes/Reloading_the_widget_during_a_call/_create/during_a_call/after_a_failed_transfer/pushes_the_event_for_the_active_call.yml:5]: I believe you can modify this without having any impact on tests at this point (e.g. `https://xxx:yyy@api...`). That's what we've done in the past to hide sensitive data. Just not sure how sensitive this particular data is.

**Pull request merged**